### PR TITLE
libsodium: Updated to latest version (1.0.11)

### PR DESCRIPTION
**Pull request by black-roland**
Maintainer: @damianorenfer
Compile tested: ar71xx OpenWTR DD/CC/BB 
Run tested: ar71xx OpenWTR DD (r48648)/CC/BB 

Description:

Updated to latest version (1.0.11)

### CODE REVIEW

**Comment by thess** [comment]
Thanks

File: libs/libsodium/Makefile [hunk 1]
```diff
@@ -52,6 +52,7 @@ endef
 
 CONFIGURE_ARGS+= \
 	--disable-ssp \
+	--without-pthreads \
```
Comment (wongsyrone) [inline libs/libsodium/Makefile]: Can you explain why disable this?

Comment (black-roland) [inline libs/libsodium/Makefile]: Because pthreads does not work with uClibc (OpenWrt CC and low).

Comment (wongsyrone) [inline libs/libsodium/Makefile]: Why not report bugs to uClibc? I don't have such issue using LEDE trunk with musl-libc.

Comment (nbd168) [inline libs/libsodium/Makefile]: A pull request for master should not disable features based on limitations in CC. It's meant to be used with LEDE/OpenWrt master.

Comment (black-roland) [inline libs/libsodium/Makefile]: @nbd168 What are my next steps? How I can split master and CC versions? Should I create PR for `for-15.05` branch?

Comment (wongsyrone) [inline libs/libsodium/Makefile]: You can select target branch when creating pull requests, and check the diff shown below before sending it.

Comment (black-roland) [inline libs/libsodium/Makefile]: As I understand CC support limited to bug-fixes and security fixes. So I made new [pull request without CC support](https://github.com/openwrt/packages/pull/3115).

**Pull request merged**